### DISCUSSION TITLE
How to run a big Javascript function filled with customized query words

### DISCUSSION THREAD

**Discussion by polt515** [gdh1995/vimium-c#704] [Q&A] [answered]
The Javascript below is suppose to work like:
```
r - go to the front page

r <search query> - search Reddit

r <r/subreddit or u/ username> - visit a user page or a subreddit

r <r/subreddit> <search query> - search a subreddit
```

```
javascript:const arg = `%s`, args = arg ? arg.split(' ') : []; let url = "https://reddit.com/";if (args[0]?.startsWith("r/") || args[0]?.startsWith("u/"))  url += args.shift() + "/";if (args.length > 0)  url += `search/?q=` + encodeURIComponent(args.join(" "));location.replace(url);
```

**Comment by gdh1995** [comment, marked as answer]
Add this `javascript:` URL as a browser bookmark, and let's say it has a name of `v_reddit`.

Add this to `Custom search engines`:
```
r: vimium://run/openBookmark#mask&title=v_reddit&#value=$S Reddit
```

And then you may use Vomnibar to type `r test` and press Enter to go to reddit.

Note: please test `r test` on a normal website tab, but not Vimium C Options, because the Options page doesn't support some types of JS syntax.

**Reply by polt515** [reply]
Whenever I try to open this InNewTab it doesn't work is, there a fix for this?

**Reply by gdh1995** [reply]
You can use your JS code to open a link in a new tab, by simulating a click event on `<a target=_blank href="...">`

**Comment by gdh1995** [comment]
It seems you commented tens of minutes ago, but I can't see the comment.

**Reply by polt515** [reply]
Everything is working perfectly Thanks for the help
